### PR TITLE
adds form config with initialModel and middleware

### PR DESCRIPTION
**Pull request by vitkon**
#### What's this PR do?
adds a config parameter to `connectForm`

#### Related issues
#6 

#### Where should the reviewer start?
FormContainer.tsx

#### How should this be manually tested?
connect a form and provide initialModel and middleware in the config object

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,8 +22,24 @@ or
...
+### `connectForm([validators], [formConfig])(WrappedComponent)`
+`form-container` exposes `connectForm` function to connect an arbitrary form component (`WrappedComponent`).
+It does not modify the component class passed to it; instead, it *returns* a new, connected component class for you to use.
+
+<a id="connect-form-arguments"></a>
+#### Arguments
+
+* [`validators: ValidationRule[]`] \(*Array*): An array of rules can be provided to validate the form model against them. Rules are executed in a sequence that is defined in the array.  
+
+* [`formConfig: IFormConfig`] \(*Object*): An object contains initial configuration for the form
```
Comment (mattdell) [inline README.md]: Is it worth mentioning this is optional?

File: src/__tests__/FormContainer.test.tsx [hunk 2]
```diff
@@ -82,6 +86,10 @@ describe('Form container', () => {
       it('should have formMethods setFieldToTouched prop', () => {
         expect(wrappedComponent.props().formMethods).toHaveProperty('setFieldToTouched');
       });
+
+      it('should get run provided middleware', () => {
+        expect(wrappedComponent.props()).toHaveProperty('bar');
+      });
```
Comment (mattdell) [inline src/__tests__/FormContainer.test.tsx]: What about...?

```
expect(wrappedComponent.state('model')).toEqual(initialModel)
```

File: README.md (hunk 1 above)
Comment (vitkon) [reply README.md]: I assumed `[]` say that

File: src/__tests__/FormContainer.test.tsx (hunk 2 above)
Comment (vitkon) [reply src/__tests__/FormContainer.test.tsx]: that's middleware, adds bar attribute to the rest of the props.

there's a test for initialModel, too

File: README.md (hunk 1 above)
Comment (mattdell) [reply README.md]: Ah. Wasn't aware of that convention, though I have a really difficult time reading method signatures like this.

lgtm

File: src/__tests__/FormContainer.test.tsx (hunk 2 above)
Comment (mattdell) [reply src/__tests__/FormContainer.test.tsx]: Sorry, I wasn't referencing the middleware test... but I don't see the test for initialModel.

Comment (vitkon) [reply src/__tests__/FormContainer.test.tsx]: this is a good assertion though
expect(wrappedComponent.props().form.model).toEqual(initialModel)

**Pull request merged**